python/tank/authentication/user.py

```diff
@@ -118,6 +126,93 @@ def impl(self):
         return self._impl
 
 
+class ShotgunSamlUser(ShotgunUser):
+    """
+    This specialized shotgun user is needed when SSO is used, as it provides
+    mechanisms
+    """
+
+    def __init__(self, impl):
+        """
+        :param impl: Internal user implementation class this class proxies.
+        """
+        super(ShotgunSamlUser, self).__init__(impl)
+        self._timer = None
+
+    def get_claims_expiration(self):
+        """
+        Obtain claims expiration for the user.
+
+        :returns: The expiration in seconds since January 1st 1970 UTC.
+        """
+        # The import must be located here. If it is at global scope, there
+        # is an import error.
+        from .shotgun_shared import saml2_sso
```

Comment:
Yes, still true. Here is the exception stack when the import is at the beginning of the file:
`
2017-07-21 13:19:51,420 [tk-desktop.bootstrap.ERROR] Unexpected exception during bootstrap.
Traceback (most recent call last):
  File "Contents/Resources/Python/bootstrap.py", line 321, in <module>
  File "/Users/hubertp/Work/Shotgun/tk-framework-desktopstartup/python/shotgun_desktop/startup.py", line 51, in <module>
    import shotgun_desktop.paths
  File "/Users/hubertp/Work/Shotgun/tk-framework-desktopstartup/python/shotgun_desktop/paths.py", line 16, in <module>
    from sgtk import LogManager
  File "/Users/hubertp/Work/Shotgun/tk-core/python/sgtk/__init__.py", line 16, in <module>
    import tank
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/__init__.py", line 58, in <module>
    from . import authentication
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/authentication/__init__.py", line 22, in <module>
    from .shotgun_authenticator import ShotgunAuthenticator
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/authentication/shotgun_authenticator.py", line 14, in <module>
    from . import user
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/authentication/user.py", line 17, in <module>
    from .shotgun_shared import saml2_sso
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/authentication/shotgun_shared/saml2_sso.py", line 27, in <module>
    from ..ui.qt_abstraction import QtCore, QtNetwork, QtGui, QtWebKit
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/authentication/ui/qt_abstraction.py", line 15, in <module>
    from ...util.qt_importer import QtImporter
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/util/__init__.py", line 20, in <module>
    from .defaults_manager import CoreDefaultsManager
  File "/Users/hubertp/Work/Shotgun/tk-core/python/tank/util/defaults_manager.py", line 17, in <module>
    from ..authentication import DefaultsManager
ImportError: cannot import name DefaultsManager
`